src/VisualStudio/Core/Def/Implementation/Workspace/SourceGeneratedFileManager.cs

```diff
@@ -57,26 +58,19 @@ internal sealed class SourceGeneratedFileManager : IRunningDocumentTableEventLis
         private readonly Dictionary<string, OpenSourceGeneratedFile> _openFiles = new();
         private readonly VisualStudioWorkspace _visualStudioWorkspace;
 
-        private readonly Dictionary<Guid, GeneratedFileDirectoryInfo> _directoryInfoOnDiskByContainingDirectoryId = new();
-
         /// <summary>
         /// When we have to put a placeholder file on disk, we put it in a directory named by the GUID portion of the DocumentId.
         /// We store the actual DocumentId (which includes the ProjectId) and some other textual information in
         /// <see cref="_directoryInfoOnDiskByContainingDirectoryId"/>, so that way we don't have to pack the information into the path itself.
         /// If we put the GUIDs and string names directly as components of the path, we quickly run into MAX_PATH. If we had a way to do virtual
         /// monikers that don't run into MAX_PATH issues then we absolutely would want to get rid of this.
         /// </summary>
-        private class GeneratedFileDirectoryInfo
-        {
-            public GeneratedFileDirectoryInfo(DocumentId documentId, Type generatorType)
-            {
-                DocumentId = documentId;
-                GeneratorType = generatorType;
-            }
+        /// <remarks>All accesses should be on the UI thread.</remarks>
+        private readonly Dictionary<Guid, SourceGeneratedDocumentIdentity> _directoryInfoOnDiskByContainingDirectoryId = new();
 
-            public DocumentId DocumentId { get; }
-            public Type GeneratorType { get; }
-        }
+        internal static readonly Option2<bool?> EnableOpeningInWorkspace =
+            new(nameof(SourceGeneratedFileManager), nameof(EnableOpeningInWorkspace), defaultValue: null,
+            storageLocations: new RoamingProfileStorageLocation("TextEditor.%LANGUAGE%.Specific.EnableOpeningSourceGeneratedFilesInWorkspaceExperiment"));
```

Comment:
should not have %LANGUAGE%, this is not a per langauge option.

Reply:
Ah, copy/paste gone bad, good catch.

Reply:
Fixed.